server/service/osquery.go

```diff
@@ -598,15 +603,15 @@ func (svc *Service) detailQueriesForHost(ctx context.Context, host *fleet.Host)
 		return nil, nil, nil
 	}
 
-	config, err := svc.ds.AppConfig(ctx)
+	features, err := svc.HostFeatures(ctx, host)
```

Comment:
@chiiph I had to make this a service method instead of the (way cooler) `Features.EnableHostUsers.Get(host)` for two reasons:

1. with the changes in the ergonomics of the feature (this being part of the team config) we _need_ to access the db, which needs to be done through a service.
2. premium features are done through a service

if you see an elegant way out of this, I'd be super onboard to trying, especially if it might also solve the whole "nested ee" problem.

Reply:
Yeah... I was wondering if (in general) some of the service methods should move to the `Host` type and so on... but it's not necessarily simple.